ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/InsertValuesExecutor.java

```diff
@@ -518,14 +544,25 @@ private byte[] serializeValue(
   private static FormatInfo addSerializerMissingFormatFields(
       final FormatInfo formatInfo,
       final String topicName,
-      final boolean isKey
+      final boolean isKey,
+      final Optional<Integer> schemaId
   ) {
     // Just add missing fields required for serialization SR formats
     final Format format = FormatFactory.fromName(formatInfo.getFormat());
     if (!format.supportsFeature(SerdeFeature.SCHEMA_INFERENCE)) {
       return formatInfo;
     }
 
+    if (schemaId.isPresent()) {
```

Comment:
should you add  the `!formatInfo.getProperties().containsKey(ConnectProperties.SCHEMA_ID)` to the condition so it does not re-build the properties againt if the ID was already specified?

Reply:
Refactored but I do have a question regarding this. Shouldn't we override it with the latest schema id? If we don't i.e. with the suggested and refactored condition, we could use the schema_id provided by the user and use that for serializing while we did all the validation against the latest schema id.

Reply:
Added a comment and will create an issue to discuss this further